bodhi/server/validators.py

```diff
@@ -214,21 +217,47 @@ def validate_nvrs(request, **kwargs):
             return
 
 
+@postschema_validator
+def validate_builds_or_from_tag_exist(request, **kwargs):
+    """
+    Ensure that at least one of the  builds or from_tags parameters exist in the request.
+
+    Args:
+        request (pyramid.request.Request): The current request.
+        kwargs (dict): The kwargs of the related service definition. Unused.
+    """
+    builds = request.validated.get('builds')
+    from_tag = request.validated.get('from_tag')
+
+    if builds is None and from_tag is None:
+        request.errors.add('body', 'builds,from_tag',
+                           "You must specify either builds or from_tag.")
+
+    if builds is not None:
+        if not isinstance(builds, list):
+            request.errors.add('body', 'builds', "The builds parameter must be a list.")
+        elif len(builds) == 0:
+            request.errors.add('body', 'builds', "You may not specify an empty list of builds.")
+
+    if from_tag is not None:
+        if not isinstance(from_tag, str):
+            request.errors.add('body', 'from_tag', "The from_tags parameter must be a string.")
+        elif len(from_tag.strip()) == 0:
+            request.errors.add('body', 'from_tag', "You may not specify an empty from_tag.")
+
+
 @postschema_validator
 def validate_builds(request, **kwargs):
     """
     Ensure that the builds parameter is valid for the request.
 
     Args:
-        request (pyramid.util.Request): The current request.
+        request (pyramid.request.Request): The current request.
         kwargs (dict): The kwargs of the related service definition. Unused.
     """
     edited = request.validated.get('edited')
     user = User.get(request.user.name)
-
-    if not request.validated.get('builds', []):
-        request.errors.add('body', 'builds', "You may not specify an empty list of builds.")
-        return
+    builds = request.validated.get('builds') or []  # cope with builds set to None
```

Comment:
I thinks these changes should also be in a separated commit from the side tag one.

Reply:
It may not look like it but this one is directly related to the from-sidetag feature, it moves checking if the list of builds is empty to the (as of yet comment-less) `else` branch in `validate_from_tag()` below.

Reply:
This would be more idiomatic:

```suggestion
    builds = request.validated.get('builds', [])
```